server/service/src/sync/test/mod.rs

```diff
@@ -28,6 +28,23 @@ impl TestSyncPullRecord {
                 r.table_name = table_name.to_owned();
                 r.record_id = id_and_data.0.to_owned();
                 r.data = id_and_data.1.to_owned();
+                r.action = SyncBufferAction::Upsert;
+            }),
+        }
+    }
+
+    fn new_pull_delete(
+        table_name: &str,
+        id: &str,
+        result_table: PullDeleteRecordTable,
+    ) -> TestSyncPullRecord {
+        TestSyncPullRecord {
+            translated_record: Some(IntegrationRecords::from_delete(id, result_table)),
+            sync_buffer_row: inline_init(|r: &mut SyncBufferRow| {
+                r.table_name = table_name.to_owned();
+                r.record_id = id.to_string();
+                r.data = "{}".to_string();
```

Comment:
i think we should have empty json in the body of delete sync buffer records, vs optional, makes it a bit easier